### PR TITLE
ci: catch error in ci for external rgw server canary test

### PR DESCRIPTION
**Pull request by parth-gr**
**Description of your changes:**

currently, we just print the error is anything fails in
script for rgw, So by that there is no panic or
failure of script if something wrong happened,

Added Explicitly forcing to catch the error from
the error message that is thrown

**Which issue is resolved by this Pull Request:**
Resolves #https://github.com/rook/rook/issues/12244

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by travisn** [comment]
nit: typo in the commit message: `Explicittly`

File: .github/workflows/canary-integration-test.yml [hunk 1]
```diff
@@ -173,17 +173,39 @@ jobs:
           rgw_endpoint=$(kubectl get service -n rook-ceph |  awk '/rgw/ {print $3":80"}')
           toolbox=$(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[*].metadata.name}')
           # pass the valid rgw-endpoint of same ceph cluster
-          timeout 15 sh -c "until kubectl -n rook-ceph exec $toolbox -- python3 /etc/ceph/create-external-cluster-resources.py --rbd-data-pool-name replicapool --rgw-endpoint $rgw_endpoint; do sleep 1 && echo 'waiting for the rgw endpoint to be validated'; done"
+          timeout 15 sh -c "until kubectl -n rook-ceph exec $toolbox -- python3 /etc/ceph/create-external-cluster-resources.py --rbd-data-pool-name replicapool --rgw-endpoint $rgw_endpoint 2> output.txt; do sleep 1 && echo 'waiting for the rgw endpoint to be validated'; done"
+          if [ -s output.txt ]; then
+            echo "script failed with stderr error"
```
Comment (travisn) [inline .github/workflows/canary-integration-test.yml:178]: Could at least some of this be factored out to a helper function to avoid duplication?

**Review by travisn — CHANGES_REQUESTED** [review]

File: .github/workflows/canary-integration-test.yml (hunk 1 above)
Comment (parth-gr) [reply .github/workflows/canary-integration-test.yml:178]: I am not sure if the function is allowed in this workflow

Comment (subhamkrai) [reply .github/workflows/canary-integration-test.yml:178]: create a function in this file https://github.com/rook/rook/blob/master/tests/scripts/github-action-helper.sh and use here

Comment (parth-gr) [reply .github/workflows/canary-integration-test.yml]: Thanks

**Pull request merged**